File: solutions/uri/1828/1828.cpp
#include <iostream>
#include <string>

int rock_paper_scissors_lizard_spock(std::string sheldons, std::string rajs) {
    if (sheldons == rajs) {
        return 0;
    }
    if ((sheldons == "tesoura" && (rajs == "papel" || rajs == "lagarto"))
        || (sheldons == "papel" && (rajs == "pedra" || rajs == "Spock"))
        || (sheldons == "pedra" && (rajs == "lagarto" || rajs == "tesoura"))
        || (sheldons == "lagarto" && (rajs == "Spock" || rajs == "papel"))
        || (sheldons == "Spock" && (rajs == "tesoura" || rajs == "pedra"))) {
        return 1;
    }
    return -1;
}

int main() {
    int t, i, result;
    std::string sheldons, rajs;

    std::cin >> t;
    for (i = 1; i <= t; i++) {
        std::cin >> sheldons >> rajs;

        result = rock_paper_scissors_lizard_spock(sheldons, rajs);

        std::cout << "Caso #" << i << ": ";
        if (result == 1) {
            std::cout << "Bazinga!";
        } else if (result == 0) {
            std::cout << "De novo!";
        } else {
            std::cout << "Raj trapaceou!";
        }
        std::cout << std::endl;
    }
    return 0;
}

File: solutions/beecrowd/1828/1828.py
import sys

input = sys.stdin.readline
output = sys.stdout.write


def rock_paper_scissors_lizard_spock_judge(sheldons_hand, rajeshs_hand):
    if sheldons_hand == rajeshs_hand:
        return 0

    if sheldons_hand == 'tesoura' and rajeshs_hand in ('papel', 'lagarto'):
        return 1
    if sheldons_hand == 'papel' and rajeshs_hand in ('pedra', 'Spock'):
        return 1
    if sheldons_hand == 'pedra' and rajeshs_hand in ('lagarto', 'tesoura'):
        return 1
    if sheldons_hand == 'lagarto' and rajeshs_hand in ('Spock', 'papel'):
        return 1
    if sheldons_hand == 'Spock' and rajeshs_hand in ('tesoura', 'pedra'):
        return 1

    return -1


t = int(input())
for case_index in range(1, t + 1):
    sheldons_hand, rajs_hand = input().split()

    result = rock_paper_scissors_lizard_spock_judge(sheldons_hand, rajs_hand)

    output(f'Caso #{case_index}: ')
    if result == 1:
        output('Bazinga!\n')
    elif result == 0:
        output('De novo!\n')
    else:
        output('Raj trapaceou!\n')
